packages/react-form/src/hooks/field/field.ts

```diff
@@ -245,6 +245,41 @@ export function asChoiceField<Value>(
   };
 }
 
+/**
+ * Converts a standard `Field<Value>` into a form that is compatible
+ * with the `<ChoiceList />` component in `@shopify/polaris`.
+ *
+ * For fields that are used by both choice components and other components, it
+ * can be beneficial to retain the original `Field<Value>` shape and convert
+ * the field on the fly for the choice component.
+ *
+ * ```typescript
+ * enum Color { Red = "red", Green = "green" }
+ *
+ * const choices = [
+ *   { label: "Red", value: Color.Red },
+ *   { label: "Green", value: Color.Green },
+ * ]
+ *
+ * const color = useField(Color.Red);
+ * return (<ChoiceList {...asChoiceList(color)} label="Color" choices={choices} />);
+ * ```
+ */
+export function asChoiceList<Value extends string | undefined | null>({
+  value,
+  onChange,
+  ...fieldData
+}: Field<Value>) {
+  return {
+    ...fieldData,
+    // eslint-disable-next-line eqeqeq
+    selected: value == undefined ? [] : [value!],
+    onChange: (selected: string[]) => {
```

Comment:
Unfortunately I think it has to be string to match the ChoiceList `onChange`? 

![image](https://user-images.githubusercontent.com/21130966/174114392-e56541b1-d19e-4eea-9fb4-416d10f3fea0.png)

Reply:
Ahh interesting, okay this is because `ChoiceList` requires a value to be set it seems, we likely want to limit `Value` to `string` only rather than `string | null | undefined` as those wouldn't be valid here

Reply:
Ohhh but we're passing it an empty array in that case, okay that should be fine then